src/katgpucbf/xbgpu/bsend.py

```diff
@@ -173,7 +173,9 @@ def timestamp(self, value: int) -> None:
         self._timestamp = value
 
     @staticmethod
-    def _inc_counters(beam_data_shape: tuple[int, int, int], beam_names: Sequence[str], future: asyncio.Future) -> None:
+    def _inc_counters(
+        beam_data_shape: tuple[int, int, int], beam_dtype: np.dtype, beam_names: Sequence[str], future: asyncio.Future
```

Comment:
Bikeshedding: I'm not sure it's really necessary to prefix everything with "beam_". This is bsend.py, so everything is implicitly beam-related.

Reply:
Too right. I think I was still in the zone of working in `test_engine`, so I felt like I needed to keep the convention between these unrelated files.